### PR TITLE
Expose read-only ManyToMany SQL metadata

### LINKED ISSUE #133: Expose read-only ManyToMany SQL metadata (junction/column names)
## Summary
It would be useful to expose read-only SQL metadata for `ManyToMany` relationships (junction table name and FK column names) via a public API.

Right now, this metadata is mostly internal/private (e.g. manager internals like `_from_col` / `_to_col`), so app code that needs occasional raw SQL has to duplicate convention-derived names.

## Real-world use case
In a real app using `sqliter-py`, I needed reporting-style queries against the M2M junction table (e.g. tag usage counts). Example shape:

```sql
SELECT tags.pk, COUNT(ideas_tags.ideas_pk)
FROM tags
LEFT JOIN ideas_tags ON ideas_tags.tags_pk = tags.pk
GROUP BY tags.pk;
```

The junction table / column names are generated by sqliter conventions, but there is no stable public way to read them for query construction.

## Why this helps
- Avoid hardcoded convention strings in app code
- Keep raw SQL aligned with ORM naming conventions
- Reduce break risk if internal naming logic changes
- Improve introspection/debugging for power users

## Suggested API direction (read-only)
Something like a public metadata object on `ManyToMany` / reverse descriptors:

- `junction_table`
- `from_column`
- `to_column`
- optional: `source_table`, `target_table`, `symmetrical`

Not asking for mutation or custom behavior here; just stable introspection for cases where raw SQL is still needed.

### PR DESCRIPTION
**Pull request by seapagan**
## Summary
- add public read-only M2M SQL metadata via `M2MSQLMetadata`
- expose `sql_metadata` on forward descriptor, reverse descriptor, manager, and prefetched wrapper
- add exhaustive tests for forward/reverse/manager paths and edge cases
- update API/docs and add a TUI ORM demo for metadata introspection
- simplify the TUI demo/docs usage to mypy-friendly forward access patterns
- add a medium-term TODO to explore explicit reverse declarations for strict typing

Closes #133 

## Summary by CodeRabbit

* **New Features**
  * Read-only SQL metadata introspection for many-to-many relationships exposed on descriptors, managers, reverse accessors and prefetched results (junction table, column/table mappings, symmetry).
  * New public M2MSQLMetadata type for structured metadata access.

* **Documentation**
  * Added API reference, guide and demo docs with examples showing how to inspect M2M SQL metadata.

* **TUI Demo**
  * New interactive demo showcasing M2M SQL metadata introspection.

* **Tests**
  * Comprehensive tests for metadata behaviour, self-references, prefetched scenarios and error handling.

### CODE REVIEW

File: sqliter/orm/m2m.py [hunk 1]
```diff
@@ -570,6 +630,7 @@ def __init__(
         self.name: Optional[str] = None
         self.owner: Optional[type] = None
         self._junction_table: Optional[str] = None
+        self._sql_metadata: Optional[M2MSQLMetadata] = None
```
[the remark this answers is not in the document]
Comment (seapagan) [reply sqliter/orm/m2m.py]: - FrozenInstanceError is available in Python 3.9 and 3.10 (not just 3.11+).
- Source of truth:
  - Python 3.9 docs include dataclasses.FrozenInstanceError
  - Python 3.10 docs include dataclasses.FrozenInstanceError
- Tests already pass in CI on 3.9/3.10 is consistent with that.
 
Also, your prompt mentions the wrong file and line number

File: sqliter/tui/demos/orm.py [hunk 2]
```diff
@@ -264,6 +264,41 @@ class User(BaseDBModel):
...
+
+    db = SqliterDB(memory=True)
+    db.create_table(Tag)
+    db.create_table(Article)
+
+    article = db.insert(Article(title="ORM Guide"))
+    python = db.insert(Tag(name="python"))
+    article.tags.add(python)
+
+    descriptor_meta = Article.tags.sql_metadata or article.tags.sql_metadata
```
[the remark this answers is not in the document]
Comment (seapagan) [reply sqliter/tui/demos/orm.py]: yeah, using `assert` is a code smell, lets not do that
